pkg/apis/datadoghq/v1alpha1/datadogmetric_types.go

```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DatadogMetricSpec defines the desired state of DatadogMetric
+type DatadogMetricSpec struct {
+	// Query is the raw datadog query
+	Query string `json:"query,omitempty"`
+	// ExternalMetricName is reversed for internal use
+	ExternalMetricName string `json:"externalMetricName,omitempty"`
+}
+
+// DatadogMetricStatus defines the observed state of DatadogMetric
+type DatadogMetricStatus struct {
+	// Conditions Represents the latest available observations of a DatadogMetric's current state.
+	// +listType=set
+	Conditions []DatadogMetricCondition `json:"conditions,omitempty"`
+	// Value is the latest value of the metric
+	Value float64 `json:"currentValue"`
+}
+
+// DatadogMetricCondition describes the state of a DatadogMetric at a certain point.
+// +k8s:openapi-gen=true
+type DatadogMetricCondition struct {
+	// Type of DatadogMetric condition.
+	Type DatadogMetricConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// Last time the condition was updated.
+	// +optional
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty"`
+	// The reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// A human readable message indicating details about the transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// DatadogMetricConditionType type use to represent a DatadogMetric condition
+type DatadogMetricConditionType string
+
+const (
+	// DatadogMetricConditionTypeActive DatadogMetric is active (referenced by an HPA), Datadog will only be queried for active metrics
+	DatadogMetricConditionTypeActive DatadogMetricConditionType = "Active"
+	// DatadogMetricConditionTypeUpdated DatadogMetric is updated
+	DatadogMetricConditionTypeUpdated DatadogMetricConditionType = "Updated"
```

Comment:
Reading in the code, I was expecting the last Update time for this

Reply:
Yeah this attribute could be seen in two ways:
- Last time the controller updated the DatadogMetric object
- Last time the metric was "Updated" from Datadog

Currently it's always "Updated" from Datadog, except in failure scenario, it's then the last time the controller updated DatadogMetric object.